babel/src/test/java/org/apache/calcite/test/BabelTest.java

```diff
@@ -189,6 +189,23 @@ private void checkInfixCast(Statement statement, String typeName, int sqlType)
         .fails("'A' is not a valid time frame");
     f2.withSql("SELECT DATE_PART(minute15, " + ts + ")")
         .ok();
+
+    final SqlValidatorFixture f3 =
```

Comment:
I think this test should be moved. When DATE_TRUNC supports custom time frames, you will no longer get the parse error "Encountered \"A\" at line 1". Add something to `SqlParserTest.testTimeUnitCodes` and something to `SqlValidatorTest.checkTimeUnitCodes` and I think you're good.